File: src/leetcode-3/go/code_test.go
package code

import "testing"

type test struct {
	s    string
	want int
}

func TestLengthOfLongestSubstring(t *testing.T) {
	tests := []test{
		{s: "abcabcbb", want: 3},
		{s: "bbbbb", want: 1},
		{s: "pwwkew", want: 3},
	}

	for _, tt := range tests {
		t.Run(tt.s, func(t *testing.T) {
			got := LengthOfLongestSubstring(tt.s)
			if got != tt.want {
				t.Errorf("LengthOfLongestSubstring(%s) = %d, want %d", tt.s, got, tt.want)
			}
		})
	}

}

File: src/leetcode-3/ts/code.test.ts
import { test, expect } from "vitest";
import { lengthOfLongestSubstring } from "./code";

test("Example 1", () => {
	const s = "abcabcbb";
	expect(lengthOfLongestSubstring(s)).eq(3);
});

test("Example 2", () => {
	const s = "bbbbb";
	expect(lengthOfLongestSubstring(s)).eq(1);
});

test("Example 3", () => {
	const s = "pwwkew";
	expect(lengthOfLongestSubstring(s)).eq(3);
});
